### PR TITLE
OCPBUGS#16215: Remove unsupported feature for VSphere storage

### PR DESCRIPTION
**Pull request by kelbrown20**
**Version(s):** 4.13+ 
**Issue:** [OCPBUGS-16215](https://issues.redhat.com/browse/OCPBUGS-16215)

**Link to docs preview:**

[Dynamic provisioning -> VMware VSphere object definition ](https://67209--docspreview.netlify.app/openshift-enterprise/latest/storage/dynamic-provisioning#vsphere-definition_dynamic-provisioning)

**QE review:**
- [x] QE has approved this change.

### CODE REVIEW

**Comment by kelbrown20** [comment]
@duanwei33 Hello! Do you think you could take a look at this bug from the QE side when you get the chance? Thank you!

File: modules/dynamic-provisioning-vsphere-definition.adoc [hunk 1]
```diff
@@ -14,14 +14,10 @@ kind: StorageClass
...
 metadata:
   name: <storage-class-name> <1>
-provisioner: kubernetes.io/vsphere-volume <2>
-parameters:
-  diskformat: thin <3>
+provisioner: csi.vsphere.vmware.com <2>
 ----
 <1> Name of the storage class. The persistent volume claim uses this storage class for provisioning the associated persistent volumes.
-<2> For more information about using VMware vSphere with {product-title},
+<2> For more information about using VMware vSphere csi with {product-title},
```
Comment (lpettyjo) [inline modules/dynamic-provisioning-vsphere-definition.adoc:20]: If "csi" refers to "Container Storage Interface", then it should be capitalized as "CSI".

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM!

**Review by kelbrown20 — COMMENTED** [review]
Testing

**Review by kelbrown20 — COMMENTED** [review]
Second test

**Review by opayne1 — APPROVED** [review]
/lgtm

**Pull request merged**